packages/shared/lib/core/wallet/actions/activities/linkTransactionsWithClaimingTransactions.ts

```diff
@@ -0,0 +1,67 @@
+import { IAccountState } from '@core/account'
+import { IProcessedTransaction } from '@core/wallet/interfaces'
+import { isOutputAsync } from '@core/wallet/utils/outputs/isOutputAsync'
+import { get } from 'svelte/store'
+import { addClaimedActivity, claimedActivities } from '../../stores'
+
+export function linkTransactionsWithClaimingTransactions(
+    transactions: IProcessedTransaction[],
+    account: IAccountState
+): IProcessedTransaction[] {
+    const resultingTransactions = []
+    const transactionsIncludedAsClaimingTransactions = []
+
+    const claimedAccountActivities = get(claimedActivities)?.[account.id]
+
+    for (const transaction of transactions.sort((t1, t2) => (t1.time > t2.time ? 1 : -1))) {
```

Comment:
Can we sort into a separate variable and then iterate on that, so its easier to read.